### DISCUSSION TITLE
Use a translation tool

### DISCUSSION THREAD

**Discussion by AndiLeni** [ExpressionEngine/ExpressionEngine#2930] [Feature Requests]
Currently, translating the control panel is a humongous amount of work, because all translations are handled in separated files.

The use of a translation tool like crowdin could make it easier for users to keep the translations up to date.

A translated backend is mandatory in many countries where english is not the primary language.

**Comment by intoeetive** [comment]
Copying over comments by @ebeauchamps from https://github.com/ExpressionEngine/ExpressionEngine/issues/736

> Right now, we still have to commit our language files to the extra-old Ellis Lab (!) repo.
> 
> I suggest that you should accept language files for 6.x to be committed to the main ExpressionEngine repo (in /user/language/)

> The nice thing with the current EllisLab repo is that the English language files are isolated form other changes. So as a translator, you can very easily follow the changes made to the lang files. If you add tag or commit comparison, you can indeed pinpoint immediately to the changes that are to be conducted to your own files. But for that, you will have to create another repo dedicated to the English language files.
> 
> Another idea is to simultaneously add or delete the language keys in the English language files and in the other locales (for those included in /user/language/). When adding them, you could have a php script calling upon the Google Translate API in order to provide at least a basic translation (75% of the time it should be acceptable).
> 
> But the difficulty arise when a word, a sentence, is edited (corrected) in the English files. As a translator, you don't want to miss this, and I am not sure GT will be of any help for you in such a case. That's why a dedicate repo is of great help: you can miss of few update (say 6.0.5 and 6.1) and still compare the latest English version (say 6.1.5) with the one you based your work on (say 6.0.0) and then find all the changes you will have to make in your "own" files.

**Reply by ebeauchamps** [reply]
Another idea:   ExpressionEngine could write a dedicated "translator" helper file dedicated to detect untranslated keys in current foreign language files (regarding the latest English files) in order to automatically add these missing keys. This helper file could also detect changes made in the latest English files (ie another sentence for the same key) in order once again to pinpoint the changes to the foreign language files.

By the way, I translated Craft CMS CP in the early years of this CMS, and they finally ditched various "translation tools" that were too costly and unable to follow all changes being made.

I am still convinced the easiest and probably cheapest way of getting and managing up-to-date translations is to watch a github repo for the diff in the various English files and then report the changes made to the foreign files.

**Reply by TomJaeger** [reply]
@ebeauchamps A lot of good thoughts here.  We realize we need to make improvements on this front.

Any chance you would be open to quick call to discuss some of these ideas?
